Macrophages/Macrophages_multiome.py

```diff
@@ -0,0 +1,746 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import re
+import math
+import os
+import seaborn as sns
+from sklearn.preprocessing import StandardScaler
+from pathlib import Path
+import scipy.stats as stats
+from scipy.stats import f_oneway
+from statsmodels.stats.multitest import multipletests
+
+# file read path 
+file_read_path = r'C:\Users\jymbc\Desktop\python files for SMAD\SMAD_online\data'
+
+# read raw protein files 
+df_lfq = pd.read_csv(f'{file_read_path}/all_proteins_after_SSnormalization.csv',index_col=0)
+
+
+def get_co_index(list1, list2):
+    set1 = set(list1)
+    set2 = set(list2)
+    intersection = set1 & set2
+    return list(intersection)
+
+def knn_imputer(df, neighbors=6):
+    '''apply KNN imputation to a dataset'''
+    from sklearn.impute import KNNImputer
+
+    # Initialize the KNNImputer
+    imputer = KNNImputer(n_neighbors=neighbors)
+
+    imputed_df = pd.DataFrame(imputer.fit_transform(df), columns=df.columns, index=df.index)
+    return imputed_df
+
+
+def iterative_imputer(df, maxiteration=10, randomstates=0):
+    from sklearn.impute import IterativeImputer
+
+    # Initialize the IterativeImputer
+    imputer = IterativeImputer(max_iter=maxiteration, random_state=randomstates)
+
+    # Create a new DataFrame with the imputed values
+    imputed_df = pd.DataFrame(imputer.fit_transform(df), columns=df.columns, index=df.index)
+    return imputed_df
+
+
+def standardscaler(df):

```

Comment:
And it's not just for this function, this applies to every function name you ever make. I like to make them short sentences, basically, that are verb-based (because functions do things).